x-pack/test/alerting_api_integration/security_and_spaces/tests/actions/builtin_action_types/swimlane.ts

```diff
@@ -0,0 +1,228 @@
+        },
+      });
+    });
+
+    it('should return unsuccessfully when passed invalid create parameters', async () => {
+      await supertest
+        .post('/api/actions/action')
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'A swimlane action',
+          actionTypeId: '.swimlane',
+          config: {
+            apiUrl: swimlaneSimulatorURL,
+          },
+          secrets: {},
+        })
+        .expect(400)
+        .then((resp: any) => {
+          expect(resp.body).to.eql({
+            statusCode: 400,
+            error: 'Bad Request',
+            message:
+              'error validating action type secrets: [routingKey]: expected value of type [string] but got [undefined]',
+          });
+        });
+    });
+
+    it('should return unsuccessfully when default swimlane url is not present in allowedHosts', async () => {
+      await supertest
+        .post('/api/actions/action')
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'A swimlane action',
+          actionTypeId: '.swimlane',
+          secrets: {},
+        })
+        .expect(400)
+        .then((resp: any) => {
+          expect(resp.body).to.eql({
+            statusCode: 400,
+            error: 'Bad Request',
+            message:
+              'error validating action type config: error configuring swimlane action: target url "https://events.swimlane.com/v2/enqueue" is not added to the Kibana config xpack.actions.allowedHosts',
+          });
+        });
+    });
+
+    it('should create swimlane simulator action successfully', async () => {
+      const { body: createdSimulatedAction } = await supertest
+        .post('/api/actions/action')
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'A swimlane simulator',
+          actionTypeId: '.swimlane',
+          config: {
+            apiUrl: swimlaneSimulatorURL,
+          },
+          secrets: {
+            routingKey: 'pager-duty-routing-key',
+          },
+        })
+        .expect(200);
+
+      simulatedActionId = createdSimulatedAction.id;
+    });
+
+    it('should handle executing with a simulated success', async () => {
+      const { body: result } = await supertest
+        .post(`/api/actions/action/${simulatedActionId}/_execute`)
+        .set('kbn-xsrf', 'foo')
+        .send({
+          params: {
+            summary: 'just a test',
+          },
+        })
+        .expect(200);
+
+      expect(proxyHaveBeenCalled).to.equal(true);
+      expect(result).to.eql({
+        status: 'ok',
+        actionId: simulatedActionId,
+        data: {
+          message: 'Event processed',
+          status: 'success',
+        },
+      });
+    });
+
+    it('should handle a 40x swimlane error', async () => {
+      const { body: result } = await supertest
+        .post(`/api/actions/action/${simulatedActionId}/_execute`)
+        .set('kbn-xsrf', 'foo')
+        .send({
+          params: {
+            summary: 'respond-with-418',
```

Comment:
this is a great way to "force errors" in the simulator, but I'm not seeing where this is handled in the simulator